services/QuillLMS/engines/evidence/app/services/evidence/vertex_ai/text_classifier.rb

```diff
@@ -65,6 +73,16 @@ def run
           .max_by { |score, i| [score, i] }
           .last
       end
+
+      private def retry_with_exceptions(max_num_retries, exception_classes)
+        num_retries = 0
+        begin
+          yield
+        rescue *exception_classes => e
+          num_retries += 1
+          num_retries <= max_num_retries ? retry : raise(e)
```

Comment:
Take or leave. `retry` is essentially a `return`, so consider coding it like a return, e.g.

```ruby
retry if num_retries <= max_num_retries
raise(e)